tests/AwsClientTest.php

```diff
@@ -340,6 +340,35 @@ public function testDoesNotSignOperationsWithAnAuthTypeOfNone()
         $client->bar();
     }
 
+    public function testSignOperationsWithAnAuthType()
+    {
+        $client = $this->createClient(
+            [
+                'metadata' => [
+                    'signatureVersion' => 'v4',
+                ],
+                'operations' => [
+                    'Bar' => [
+                        'http' => ['method' => 'POST'],
+                        'authtype' => 'v4-unsigned-body',
+                    ],
+                ],
+            ],
+            [
+                'handler' => function (
+                    CommandInterface $command,
+                    RequestInterface $request
+                ) {
+                    foreach (['Authorization','X-Amz-Content-Sha256', 'X-Amz-Date'] as $signatureHeader) {
+                        $this->assertTrue($request->hasHeader($signatureHeader));
```

Comment:
Besides verify the header exists, perhaps also verifys it's using 'UNSIGNED-PAYLOAD'